File: Easy/1925_count_square_sum_triples/solution.py
class Solution:
    def countTriples(self, n: int) -> int:
        cnt = 0
        for a in range(1, n):
            for b in range(a, n):
                x = math.sqrt(a**2 + b**2)
                if x > n:
                    break
                if (x*10)%10 == 0:
                    cnt += 2

        return cnt

File: Easy/1925_count_square_sum_triples/solution.cpp
#include <cmath>
#include <iomanip>
class Solution {
public:
    int countTriples(int n) {
        int cnt = 0;
        for (int a=1; a<=n; a++){
            for (int b=a; b<=n; b++){
                int c_squared = a*a + b*b;
                int c = static_cast<int>(std::sqrt(c_squared));
                if (c<=n && c*c==c_squared){
                    if (a != b){
                        cnt += 2;
                    }
                    else {
                        cnt += 1;
                    }
                }
            }
        }
        return cnt;
    }
};
